### PR TITLE
Fixed bug when training

### PR DESCRIPTION
**Pull request by axel012**
Learning rate never being initialized, that will fix your mnist project too. 
Stay awesome!

### CODE REVIEW

File: nn.js [hunk 1]
```diff
@@ -44,8 +44,8 @@ function mutate(x) {
 }
 
 // Neural Network constructor function
-function NeuralNetwork(inputnodes, hiddennodes, outputnodes, activation) {
-
+function NeuralNetwork(inputnodes, hiddennodes, outputnodes, learningRate) {
```
Comment (shiffman) [inline nn.js]: shouldn't this be `function NeuralNetwork(inputnodes, hiddennodes, outputnodes, learningRate, activation) {`?

Comment (axel012) [inline nn.js]: yep, that will be better!, hope it fix your issues with the examples!

**Comment by axel012** [comment]
I think i didn't explain what i fix, so here is what should be changed:
Line 132->   gradient_output.multiply(this.lr); 
Line 138->  gradient_hidden.multiply(this.lr);
This.lr is never initialized, you are multipliying your results by undefined, that will give  NaN values in your wih and who matrixes, in other words weights will be broken.

**Comment by shiffman** [comment]
Thank you so much for the help! I made some other fixes so this is now covered in 8e2c5d13e155c90a8d29561c8d3deee68b30df44.

**Pull request closed without merging**